lib/types/p2p.ts

```diff
@@ -9,6 +9,7 @@ export type Address = {
 export type NodeConnectionInfo = {
   nodePubKey: string;
   addresses: Address[];
+  lastAddress: Address | undefined;
```

Comment:
Rather than explicitly put `| undefined`, just call this `lastAddress?` to be consistent with our other type definitions.